first_hour_on_terra/working_with_data_collections.ipynb

```diff
@@ -0,0 +1,476 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "### Create a new workspace\n",
+    "\n",
+    "In order to create a data collection, you must first create a new workspace. Run the cell below to create a new workspace. The output should resemble:\n",
+    "\n",
+    "```\n",
+    "Workspace successfully created.\n",
+    "ID:                <GOOGLE_PROJECT_ID>-dc-ws\n",
+    "Name:              <TERRA_USER_EMAIL>-Data\n",
+    "Description:       A new workspace which I will transform into a data collection.\n",
+    "Cloud Platform:    GCP\n",
+    "Google project:    <GOOGLE_PROJECT_ID>\n",
+    "Cloud console:     https://console.cloud.google.com/home/dashboard?project=terra-vpp-arctic-mustard-2133\n",
+    "Properties:\n",
+    "  terra-type: workspace\n",
+    "Created:           YYYY-MM-DD\n",
+    "Last updated:      YYYY-MM-DD\n",
+    "# Resources:       0\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "buttonOutput = 'Please provide appropriate values in the input boxes.'\n",
+    "\n",
+    "input_workspace_id = widgets.Text(\n",
+    " placeholder=\"<WORKSPACE_ID>\",\n",
+    " description=\"Workspace ID:\",\n",
+    " style=input_style\n",
+    ")\n",
+    "output_workspace_id = widgets.Text()\n",
+    "input_description = widgets.Text(\n",
+    " placeholder=\"<DESCRIPTION>\",\n",
+    " description=\"Description:\",\n",
+    " style=input_style\n",
+    " )\n",
+    "input_name = widgets.Text(\n",
+    " placeholder=\"<RESOURCE_NAME>\",\n",
+    " description=\"Resource Name:\",\n",
+    " style=input_style\n",
+    " )\n",
+    "display(input_workspace_id)\n",
+    "display(input_description)\n",
+    "display(input_name)\n",
+    "\n",
+    "def bind_input_to_output(e):\n",
+    "    output_workspace_id.value = input_workspace_id.value\n",
+    "\n",
+    "# define a function for the button to call\n",
+    "def button_click_event(b):\n",
+    "    with output:\n",
+    "        global buttonOutput\n",
+    "        terraCommand = f\"terra workspace create --id={input_workspace_id.value} --description=\\\"{input_description.value}\\\" --name={input_name.value}\"\"\".format(input_workspace_id.value,input_description.value,input_name.value)       \n",
+    "        print(\"Running command to create workspace...\")\n",
+    "        print(terraCommand)\n",
+    "        result = subprocess.run([\"terra\",\"workspace\",\"create\",f\"--id={input_workspace_id.value}\",f\"--description=\\\"{input_description.value}\\\"\",f\"--name={input_name.value}\"],\n",
```

Comment:
What happens if this command fails? Say, I give a workspace ID that already exists.

Reply:
I have now updated both widgets to print the success message (from stdout) if the command succeeds, but to otherwise print the error message (from stderr) so that users can learn from the error message what went wrong.